From 8a48c61c61df574bbda888a5a78dcedc37afc51d Mon Sep 17 00:00:00 2001
From: jerrita <je5r1ta@icloud.com>
Date: Mon, 15 Jan 2024 15:43:16 +0800
Subject: [PATCH] add options for miniupnpd-nftables

---
 .../modules/services/networking/miniupnpd.nix | 20 +++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/nixos/modules/services/networking/miniupnpd.nix b/nixos/modules/services/networking/miniupnpd.nix
index 116298dc6b1db1..8f55e3ac1b21e8 100644
--- a/nixos/modules/services/networking/miniupnpd.nix
+++ b/nixos/modules/services/networking/miniupnpd.nix
@@ -14,8 +14,8 @@ let
     '') cfg.internalIPs}
 
     ${lib.optionalString (firewall == "nftables") ''
-      upnp_table_name=miniupnpd
-      upnp_nat_table_name=miniupnpd
+      upnp_table_name=${cfg.upnpTableName}
+      upnp_nat_table_name=${cfg.upnpNatTableName}
     ''}
 
     ${cfg.appendConfig}
@@ -55,6 +55,22 @@ in
         '';
       };
 
+      upnpTableName = mkOption {
+        default = "miniupnpd";
+        type = types.str;
+        description = lib.mdDoc ''
+          upnp_table_name for nftables.
+        '';
+      };
+
+      upnpNatTableName = mkOption {
+        default = "miniupnpd";
+        type = types.str;
+        description = lib.mkDoc ''
+          upnp_nat_table_name for nftables.
+        '';
+      };
+
       appendConfig = mkOption {
         type = types.lines;
         default = "";
